addons/mail/data/mail_templates_email_layouts.xml

```diff
@@ -8,17 +8,19 @@
 </head>
 <body style="font-family:Verdana, Arial,sans-serif; color: #454748;">
 <t t-set="subtype_internal" t-value="subtype and subtype.internal"/>
+<t t-set="show_header" t-value="has_button_access or email_show_header"/>
```

Comment:
Here, I just applied the logic Luc wants but, maybe, we should consider that it's the caller responsibility to show/hide the header/footer?

Like: When I set `email_show_footer` to true, I expect the mail to have a footer. I'm not sure it's a good idea to add other conditions hidden in the template that would make the option pointless.

What do you think?

Reply:
I just have one question then, do we really need the `email_show_footer` context key?
I mean, we only show the footer if the header is present and the user is internal. Maybe this is sufficient and we have no need for such a possibility?
I don't see the use case of showing only the footer in the mail, if you have one feel free to explain it.

Reply:
It's because we want to show the footer for specific flow only. The footer contains an advertisement for Odoo ("Powered by Odoo") and we don't want to include that in every mail exchange. We now want to show that only when the user is about to close a deal or something (when sending a invoice, a purchase/sale order, etc).

The footer is now hidden by default and you can set the context key `email_show_footer` to enable it. When enabling it, the system will include the footer in the email if it makes sens, i.e: when the sender is an internal user and there is an action button included in the email.

Here, I'm not sure it's a good idea to add additional hidden conditions in the template. People may not understand why the footer is sometimes not included in the email despite setting `email_show_footer` to `True`.

Reply:
**Let's maybe see that at the review?**
I think we could enable/disable the header/footer in `_notify_get_recipients_groups` as we do with `has_button_access`.

Reply:
This is a possibility but I am not a big fan of it. This only resolves part of the issue because you have all the custom overrides done in a lot of modules. What will you do about those? 
What is the default behavior without the key present?
This solution raises even more questions in my opinion.

Reply:
Ok, to avoid the overrides, we can set the variables `email_show_header`/`email_show_footer` via the context for the flows requiring a header and a footer (as currently done on the PR).

Reply:
I would do it in the method preparing rendering variables. That way the condition is inside the code, not in the template, which is therefore more straightforward (easier to override code than templates). Template receives instructions and apply them, role of the method rendering it to give the right values.

Aka: would move that in '_notify_by_email_prepare_rendering_context' .

Reply:
Mmmh, somewhere I did modify some bits linked to that in master, while preparing other stuff for mail : https://github.com/odoo/odoo/commit/5c82e0b754b74c4b91137b9aa7fe74f8d2d3a041 . I directly update 'show_unfollow' button in '_notify_get_classified_recipients_iterator' based on specific recipients, just before the rendering. That was the condition is not in the template. I suppose we can do the same here.

Reply:
Ah but not sure you have all variables in 18, like 'recipients_data' ... to check, maybe it wont' be doable like that in stable.